src/JustSaying/AwsTools/QueueCreation/AmazonQueueCreator.cs

```diff
@@ -33,47 +33,70 @@ public AmazonQueueCreator(IAwsClientFactoryProxy awsClientFactory, ILoggerFactor
 
-            var queue = await EnsureQueueExistsAsync(region, queueConfig).ConfigureAwait(false);
+            var queueWithStartup = EnsureQueueExists(region, queueConfig);
 
-            if (TopicExistsInAnotherAccount(queueConfig))
+            async Task StartupTask()
             {
-                var arnProvider = new ForeignTopicArnProvider(regionEndpoint, queueConfig.TopicSourceAccount, queueConfig.PublishEndpoint);
+                await queueWithStartup.StartupTask.ConfigureAwait(false);
+                var queue = queueWithStartup.Queue;
+                if (TopicExistsInAnotherAccount(queueConfig))
+                {
+                    var arnProvider = new ForeignTopicArnProvider(regionEndpoint,
+                        queueConfig.TopicSourceAccount,
+                        queueConfig.PublishEndpoint);
 
-                var topicArn = await arnProvider.GetArnAsync().ConfigureAwait(false);
-                await SubscribeQueueAndApplyFilterPolicyAsync(snsClient, topicArn, sqsClient, queue.Uri, queueConfig.FilterPolicy).ConfigureAwait(false);
-            }
-            else
-            {
-                var eventTopic = new SnsTopicByName(queueConfig.PublishEndpoint, snsClient, serializationRegister, _loggerFactory, messageSubjectProvider);
-                await eventTopic.CreateAsync().ConfigureAwait(false);
+                    var topicArn = await arnProvider.GetArnAsync().ConfigureAwait(false);
+                    await SubscribeQueueAndApplyFilterPolicyAsync(snsClient,
+                        topicArn,
+                        sqsClient,
+                        queue.Uri,
+                        queueConfig.FilterPolicy).ConfigureAwait(false);
+                }
+                else
+                {
+                    var eventTopic = new SnsTopicByName(queueConfig.PublishEndpoint,
+                        snsClient,
+                        serializationRegister,
+                        _loggerFactory,
+                        messageSubjectProvider);
+                    await eventTopic.CreateAsync().ConfigureAwait(false);
 
-                await SubscribeQueueAndApplyFilterPolicyAsync(snsClient, eventTopic.Arn, sqsClient, queue.Uri, queueConfig.FilterPolicy).ConfigureAwait(false);
+                    await SubscribeQueueAndApplyFilterPolicyAsync(snsClient,
+                        eventTopic.Arn,
+                        sqsClient,
+                        queue.Uri,
+                        queueConfig.FilterPolicy).ConfigureAwait(false);
 
-                await SqsPolicy.SaveAsync(eventTopic.Arn, queue.Arn, queue.Uri, sqsClient).ConfigureAwait(false);
+                    await SqsPolicy
+                        .SaveAsync(eventTopic.Arn, queue.Arn, queue.Uri, sqsClient)
+                        .ConfigureAwait(false);
+                }
             }
 
-            return queue;
+            return new QueueWithAsyncStartup<SqsQueueByName>(StartupTask(), queueWithStartup.Queue);
```

Comment:
I'd add a comment here making it clear that the task from `StartupTask()` is not awaited on purpose.

Reply:
Done!